src/unstable_opts.rs

```diff
@@ -0,0 +1,99 @@
+// src/bin/tectonic.rs -- Command-line driver for the Tectonic engine.
+// Copyright 2020 the Tectonic Project
+// Licensed under the MIT License.
+
+//! Unstable options for the Tectonic engine.
+//!
+//! This is similar to the -Z options on rustc - they're unstable options that are not guaranteed
+//! to be reliable or very polished. In particular, many of these prevent the build from being
+//! reproducible.
+
+use crate::errors::{Error, Result};
+use std::default::Default;
+use std::str::FromStr;
+
+const HELPMSG: &str = r#"Available unstable options:
+
+    -Z help                     Lists all unstable options
+    -Z min-crossrefs=<num>      Equivalent to bibtex's -min-crossrefs flag - "include after <num>
+                                    crossrefs" [default: 2]
+    -Z paper-size=<spec>        Change the default paper size [default: letter]
+    -Z shell-escape             Enable \write18
```

Comment:
My biggest comment: I do not want to expose a `shell-escape` option until it is actually implemented. I always find it *super* frustrating when a piece of software gives the impression that it supports something, then you try to use the feature and it doesn't actually work. We should keep the infrastructure that you've put together so far, but make sure not to give the impression that \write18 is available yet.

Reply:
Sounds good. I've removed it from the help and UnstableArg parsing.